Automattic/newspack-plugin
# fix(GA): update spelling of referrer param

### All Submissions:






### Changes proposed in this Pull Request:



Closes `0/1206331646877116/1206412046494468/f`.

### How to test the changes in this Pull Request:

1. Make sure your site is [connected to GA](https://github.com/Automattic/newspack-plugin/tree/trunk/includes/data-events/connectors/ga4#setting-up)
2. Trigger a user registration event by registering for a new user by donating as a new unsubscribed user
3. In GA reports > realtime tab, look for the `event count by event name` card
4. Look for the `np_reader_registered` and `np_donation_new` events
5. Clicking each of these, confirm you see a `referrer` field. For example:
![Screenshot 2024-02-09 at 14 18 23](https://github.com/Automattic/newspack-plugin/assets/17905991/02924c7c-27e4-4766-8eef-c9b6c71c20eb)
6. There are two other events which pass along a referrer param to GA: `np_newsletter_subscribed` and `np_gate_interaction`. Verify these as well by triggering their respective actions as a subscriber
7. Have a look through the code changes to verify nothing important was missed. Note that some instances of `referer` will remain in the code as some wordpress functions use this spelling.

Note: I'm not sure if it will be a problem for publishers if we completely remove the `referer` field in favor of `referrer`. If it will be, I can re-add `referer` alongside `referrer`.

### Other information:




---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206412046494468
